scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -241,6 +241,255 @@ fun Bitmap.zoom(
     return this.fragment(fromSegments, toSegments, toSize)
 }
 
+fun Bitmap.zoomOffset(
+    centerSize: Size,
+    previewSize: Size,
+    cardFinder: Rect,
+    toCenterDimension: Int,
+    toBorderWidth: Int
+): Bitmap {
+    require(cardFinder.left > 0 && cardFinder.right < previewSize.width && cardFinder.top > 0 && cardFinder.bottom < previewSize.height) {
+        "Card viewfinder window extends beyond preview boundary"
+    }
+    val newLeft = cardFinder.left * this.width / previewSize.height
+    val newTop = cardFinder.top * this.height / previewSize.height
+    val newRight = cardFinder.right * this.width / previewSize.width
+    val newBottom = cardFinder.bottom * this.height / previewSize.height
+
+    val scaledCardFinderRect = Rect(newLeft, newTop, newRight, newBottom)
+
+    val x = scaledCardFinderRect.centerX()
+    val y = scaledCardFinderRect.centerY()
+
+    require(x + centerSize.width < this.width && x - centerSize.width > 0 && y + centerSize.height < this.height && y - centerSize.height > 0) {
+        "Card viewfinder is to close to the edge of the screen"
+    }
+
+    require(toCenterDimension > 0 && toBorderWidth > 0) {
```

Comment:
this can probable be `require(toCenterDimension >= 0 && toBorderWidth >= 0) {`. This way, we can use this function to crop out the center and border if we so desire.